xpdview/callbacks.py

```diff
@@ -0,0 +1,40 @@
+from bluesky.callbacks.broker import BrokerCallbackBase
+from .waterfall import Waterfall
+import matplotlib.pyplot as plt
+
+
+class LiveWaterfall(BrokerCallbackBase):
+    """
+    Stream 2D images in a cross-section viewer.
+
+    Parameters
+    ----------
+    field : string
+        name of data field in an Event
+    """
+
+    def __init__(self, x_name, y_name, db=None, units=None):
+        import matplotlib.pyplot as plt
+        super().__init__((x_name, y_name,), db=db)
+        self.db = db
+        self.x_name = x_name
+        self.y_name = y_name
+        self.units = units
+        self.fig = plt.figure()
```

Comment:
suggest to use 
````
from matplotlib.figure import Figure
self. fig = Figure()
````
``plt`` comes with its own complex backends to identify the states which sometimes mess things up.

Wisdom from @tacaswell